### PR TITLE
Add regexes

### PR DESCRIPTION
**Pull request by 0xlynett**
A set of regexes that might be useful for devs. It's probably more restrictive than the official docs, mainly because the official docs don't really put any restrictions on names as far as I know.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,6 +47,14 @@
 * [Discord Bot API for processes](https://github.com/kinode-dao/discord_api)
 * [LLM API](https://github.com/kinode-dao/llm), for Groq, OpenAi, as well as llama.cpp.
 
+### Useful Regexes
+* full process addresses: `^([a-z0-9-_.]+)@([a-z0-9\-_]+):([a-z0-9-_]+):([a-z0-9-_.]+)#([a-z]+)$`
+* package IDs: `^([a-z0-9-_]+):([a-z0-9-_.]+)$`
+* process IDs: `^([a-z0-9-_]+):([a-z0-9-_]+):([a-z0-9-_.]+)$`
+* KNS names: `^([a-z0-9-_]+)\.([a-z]+)$`
+
+To ensure forward compatibility, these regexes do not require any TLD such as `.os` or `.eth` (except for the KNS name regex). Additionally, only alphanumeric characters, `-` and `_` are allowed within names. [Test them on RegExr.](https://regexr.com)
```
Comment (nick1udwig) [inline README.md]: Let's change this to one sentence per line.

File: README.md [hunk 2]
```diff
@@ -46,6 +47,14 @@
 * [Discord Bot API for processes](https://github.com/kinode-dao/discord_api)
 * [LLM API](https://github.com/kinode-dao/llm), for Groq, OpenAi, as well as llama.cpp.
 
+### Useful Regexes
+* full process addresses: `^([a-z0-9-_.]+)@([a-z0-9\-_]+):([a-z0-9-_]+):([a-z0-9-_.]+)#([a-z]+)$`
```
Comment (nick1udwig) [inline README.md]: I think there is a typo here. It doesn't work as expected for, e.g.,
```
foo.os@bar:baz:template.os
```
but does work for
```
foo.os@bar:baz:template#os
```

Typo?

File: README.md [hunk 3]
```diff
@@ -46,6 +47,14 @@
...
+* package IDs: `^([a-z0-9-_]+):([a-z0-9-_.]+)$`
+* process IDs: `^([a-z0-9-_]+):([a-z0-9-_]+):([a-z0-9-_.]+)$`
+* KNS names: `^([a-z0-9-_]+)\.([a-z]+)$`
```
Comment (nick1udwig) [inline README.md]: I have mixed feelings about `KNS names` splitting the name & TLD while the previous return them together as a single unit. At a minimum we should have a brief description of the expected behavior, maybe an example for each.

**Review by nick1udwig — CHANGES_REQUESTED** [review]
Formatting nit, possible typo, and an overall suggestion. It might be easier to read/present the regexes + what they are + example input & output in a table. Maybe something like

Type | Regex | Example input | Example output | Description
------ | --------- | ----------------- | ------------------ | ------------------
...


```
Type | Regex | Example input | Example output | Description
------ | --------- | ----------------- | ------------------ | ------------------
...
```
Alternative formats also acceptable!

Sorry for the slow review.

File: README.md (hunk 2 above)
Comment (0xlynett) [reply README.md]: ah yeah, i must've typoed it. i'm also working on a thing that involves a custom address format based on the kinode address format, that is probably why

**Review by nick1udwig — APPROVED** [review]
Ok this is looking really good. Two last change requests, then feel free to merge.

1. I can wrap my head around KNS names parsing to two parts while the others parse to one (i.e. KNS: `template.os` -> `[template, os]`). But we still have `full process addresses` splitting like KNS. We should have the three longer ones do the same behavior wrt not splitting the publisher.
2. Let's change the order: swap package id and process id, so we have a consistently decreasing hierarchy rather than a mostly decreasing hierarchy.

**Comment by nick1udwig** [comment]
Thank you for the PR!

**Pull request merged**